NBitcoin/IpExtensions.cs

```diff
@@ -255,6 +302,30 @@ public static string ToEndpointString(this EndPoint endpoint)
 			return endpoint.ToString();
 		}
 
+		public static string GetStringAddress(this EndPoint endPoint)
+		{
+			return endPoint switch
+			{
+				IPEndPoint ip => ip.Address.ToString(),
+				DnsEndPoint dns => dns.Host,
+				_ => throw new ArgumentException($"Unknown {nameof(EndPoint)} type.") 
+			};
+		}
+
+		public static bool IsEqualTo(this EndPoint endPoint1, EndPoint endPoint2)
+		{
+			if (Object.ReferenceEquals(endPoint1, endPoint2))
+				return true;
+
+			if (endPoint1 is IPEndPoint ip1 && endPoint2 is IPEndPoint ip2)
+				return ip1.Address.EnsureIPv6().Equals(ip2.Address.EnsureIPv6()) && ip1.Port == ip2.Port;

```

Comment:
`EnsureIPv6` throws exception `if (address.AddressFamily != AddressFamily.InterNetwork) && if (address.AddressFamily == AddressFamily.InterNetworkV6)`: https://github.com/nopara73/NBitcoin/blob/4f9ee43582e5631c7a687cab581b98b46461fa72/NBitcoin/Utils.cs#L573-L578

Reply:
There are only two types of IP addresses: IPv4 and IPv6. .NET IPAddress has that weird concept of `AddressFamily` and a property that could be, in theory, one of many values however, in real world there is no way to set that `AddressFamily` to anything other than `InterNetwork` and `InterNetworkV6`.

The `MapToIPv6` extension method is smart enough to deal with the question: what is someone manipulates the internal state of and IPAddress using reflection?

Reply:
I reviewed the code IPAddress source code and it is not possible to have anything other than `InterNetwork` and `InterNetworkV6`. Anyway, i will try to find a way to make these line better.

Reply:
ACK